libraries/chain/htlc_evaluator.cpp

```diff
@@ -111,10 +111,11 @@ namespace graphene {
 
       void_result htlc_redeem_evaluator::do_apply(const htlc_redeem_operation& o)
       {
-         db().adjust_balance(htlc_obj->transfer.to, asset(htlc_obj->transfer.amount, htlc_obj->transfer.asset_id) );
+         const auto amount = asset(htlc_obj->transfer.amount, htlc_obj->transfer.asset_id);
+         db().adjust_balance(htlc_obj->transfer.to, amount);
          // notify related parties
-         htlc_redeemed_operation virt_op( htlc_obj->id, htlc_obj->transfer.from, htlc_obj->transfer.to,
-               o.redeemer, asset(htlc_obj->transfer.amount, htlc_obj->transfer.asset_id ) );
+         htlc_redeemed_operation virt_op( htlc_obj->id, htlc_obj->transfer.from, htlc_obj->transfer.to, o.redeemer,
+               amount, htlc_obj->conditions.hash_lock.preimage_hash, htlc_obj->conditions.hash_lock.preimage_size );
```

Comment:
If the HTLC was redeemed by `to`, she won't get notified by the virtual `htlc_redeemed_operation`, thus it's still inconvenient for her since the new added fields are not in `htlc_redeem_operation`. In addition, the redeemer won't get notified with the virtual operation. IMHO it's better to simply notify all parties in `get_impacted_account_visitor::operator()( const htlc_redeemed_operation& op )`. Need to take care of account history numbering if to take this approach.